### ISSUE TITLE
Changed name case does not apply when running dump-autoload

### ISSUE THREAD

**Issue by MageParts** [composer/composer#10133]
The information requested by the template could either not be provided, or it was irrelevant for this issue, so I omitted it.

I had a package with an autoload section like this:

```
"autoload": {
    "psr-4": {
      "AcmeWhatever\\Core\\": "src"
    }
  }
```

My classes wouldn't load, and I noticed I had camel cased "AcmeWhatever", but used the namespace "Acmewhatever" in my classes.

I fixed this in the autoload definition, so it read:

```
"autoload": {
    "psr-4": {
      "Acmewhatever\\Core\\": "src"
    }
  }
```

I cleared my cache, I generated my autoloaders, but the change didn't take effect. The name persisted as camel cased in the autoloaders. After fixing that manually my classes loaded as expected.

Composer version 2.1.8

**Comment by stof** [comment]
is that for your root package or for a dependency ? If for a dependency, how did you perform the change ?

**Comment by MageParts** [comment]
@stof It's for a package I'm currently developing locally, I've required it in my base composer.json file using a path mapping. To elaborate further, I'm developing a Prestashop module for a client, I created a directory outside the website root directory, added my composer.json file there and made the mistake I described above. Then I required the module from the website root directory, and then I noticed my mistake. Not sure any of that helps though?
